### PR TITLE
71 - feat: openai image creation fee

### LINKED ISSUE #71: Image creation fee
Enforce a 1 TNB per image creation fee.

For example, this operation would cost 5 TNB.

Screen Shot 2024-03-26 at 8 29 54 PM

### PR DESCRIPTION
**Pull request by MAbdurrehman1**
Closes #71

### CODE REVIEW

File: thenewboston/cores/utils/core.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.conf import settings
+
+from thenewboston.cores.models import Core
+
+
+def get_default_core():
+    """
```
Comment (dmugtasimov) [inline thenewboston/cores/utils/core.py:7]: The doc-string does not add any useful information aside what we already know from the function name, so let' remove it

**Review by dmugtasimov — COMMENTED** [review]
I reviewed only parts I am aware about.

File: thenewboston/cores/utils/core.py (hunk 1 above)
Comment (MAbdurrehman1) [reply thenewboston/cores/utils/core.py:7]: > The doc-string does not add any useful information aside what we already know from the function name

Agreed. Let me remove it.
Thanks!

**Comment by MAbdurrehman1** [comment]
@buckyroberts Can you please have another look?

File: thenewboston/art/views/openai_image.py [hunk 2]
```diff
@@ -15,18 +20,51 @@
...
+
+            description = serializer.validated_data['description']
+            quantity = serializer.validated_data['quantity']
+            total_image_creation_fee = OPENAI_IMAGE_CREATION_FEE * quantity
+
+            wallet = get_default_wallet(request.user)
+            if not wallet:
+                raise Exception(f'Core {settings.DEFAULT_CORE_TICKER} wallet not found.')
+
+            self.has_sufficient_balance_for_image_creation(wallet, total_image_creation_fee)
```
Comment (buckyroberts) [inline thenewboston/art/views/openai_image.py:36]: These checks to make sure the wallet exists and that the user has a sufficient balance for image creation need to be moved to the serializer. That is where we will handle all validation. Similar to this: https://github.com/thenewboston-developers/thenewboston-Backend/blob/master/thenewboston/exchange/serializers/exchange_order.py#L75-L88

File: thenewboston/art/views/openai_image.py [hunk 3]
```diff
@@ -15,18 +20,51 @@
...
-                quantity=validated_data['quantity'],
+                prompt=description,
+                quantity=quantity,
             )
+
+            self.charge_image_creation_fee(wallet, total_image_creation_fee)
+
             # TODO(dmu) LOW: Consider using status.HTTP_201_CREATED instead
             return Response(response.dict(), status=status.HTTP_200_OK)
         except Exception as e:
```
Comment (buckyroberts) [inline thenewboston/art/views/openai_image.py:47]: There should be no `try/Except` needed as all data should be valid.

File: thenewboston/art/views/openai_image.py (hunk 2 above)
Comment (MAbdurrehman1) [reply thenewboston/art/views/openai_image.py:36]: @buckyroberts Thanks for pointing this out. will do that

File: thenewboston/art/views/openai_image.py (hunk 3 above)
Comment (MAbdurrehman1) [reply thenewboston/art/views/openai_image.py:47]: Sure!

**Pull request merged**